prow/flagutil/controller_manager.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"time"
+)
+
+type ControllerManagerOptions struct {
+	TimeoutListingProwJobs        time.Duration
+	TimeoutListingProwJobsDefault time.Duration
+}
+
+func (o *ControllerManagerOptions) AddFlags(fs *flag.FlagSet) {
+	fs.DurationVar(&o.TimeoutListingProwJobs, "timeout-listing-prowjobs", o.TimeoutListingProwJobsDefault, "Timeout for listing prowjobs in seconds.")
```

Comment:
This is not "in seconds".

Reply:
Will fix

Reply:
Thinking about this a little more, it feels very odd that we'll be telling our users to specify this as a pure time.Duration (nanosecond precision) value at runtime. IOW it feels wrong to expect users to say `--timeout-listing-prowjobs=10000000000` to mean 10 seconds. I think I should use a different type (probably just uint) and deal with the type casts.

Cf. https://pkg.go.dev/time#Duration

Reply:
Fortunately that is not how `DurationVar` works and we are not expecting our users to specify a nonsecond precision value, but rather a human readable duration: https://pkg.go.dev/flag#FlagSet.DurationVar

Reply:
TIL. I had a sneaky feeling that there just had to be a better way --- and I'm glad my uint idea is completely unnecessary!

Reply:
Done